### DISCUSSION TITLE
does prisma support lazy loading by default?

### DISCUSSION THREAD

**Discussion by softmarshmallow** [prisma/prisma#3899] [Q&A] [answered]
does prisma support lazy loading loke typeorm does?

**Comment by ryands17** [comment, marked as answer]
Hi 👋 

Prisma by default have lazy loading of queries as it only loads the relations when you specify either `select` or `include`. It also has a fluent APi that you can use to fetch the relations when required so in my opinion, lazy loading isn't required in Prisma. You can read more about relation queries [here](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/relation-queries).

Also it would be great if you could you explain your use case for lazy loading queries that you have faced while using Prisma.

**Reply by softmarshmallow** [reply]
That cleared me up! thanks

**Reply by rickerp** [reply]
I know it's a bit far away of the topic but, what about being able to access the relationship models like __virtual__ column that only when accessed it would query the database (lazy load). A little bit like other ORMs relationships (sqlalchemy p.e.)

**Reply by ryands17** [reply]
We don't support virtual columns yet, only relation columns and they are only fetched when you include the relation: https://github.com/prisma/prisma/issues/3394

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
